uitools/cpp/Esri/ArcGISRuntime/Toolkit/UtilityNetworkTraceController.cpp

```diff
@@ -0,0 +1,1026 @@
+  QHash<FeatureLayer*, QList<Feature*>> layerToFeatures;
+  for (const auto f : *featuresList)
+  {
+    auto featureLayer = static_cast<FeatureLayer*>(f->featureTable()->layer());
+
+    auto findLayer = layerToFeatures.find(featureLayer);
+    if (findLayer == std::end(layerToFeatures))
+    {
+      layerToFeatures[featureLayer] = {f};
+    }
+    else
+    {
+      findLayer.value().append(f);
+    }
+  }
+
+  const auto layerKeys = layerToFeatures.keys();
+  for (const auto lyr : layerKeys)
+  {
+    // this is where the functionality takes place
+    lyr->unselectFeatures(layerToFeatures[lyr]);
+  }
+}
+
+void UtilityNetworkTraceController::onTraceCompleted()
+{
+  m_traceResults = m_selectedUtilityNetwork->traceResult();
+
+  QList<UtilityElement*> allElements;
+  bool hasGeometryTraceResult{false};
+
+  for (const auto result : *m_traceResults)
+  {
+    const auto type = result->traceResultObjectType();
+
+    switch (type)
+    {
+      case UtilityTraceResultObjectType::UtilityElementTraceResult:
+      {
+        // Trace completed with Element Result
+        setIsResetResultsEnabled(true);
+        auto elementResult = static_cast<UtilityElementTraceResult*>(result);
+        allElements.append(elementResult->elements());
+        break;
+      }
+      case UtilityTraceResultObjectType::UtilityFunctionTraceResult:
+      {
+        // Trace completed with Function Result
+        setIsResetResultsEnabled(true);
+        const auto functionResult = static_cast<UtilityFunctionTraceResult*>(result);
+        const auto outputList = functionResult->functionOutputs();
+
+        for (const auto o : outputList)
+        {
+          m_functionResults->addFunctionResult(new UtilityNetworkFunctionTraceResult(o->function()->networkAttribute()->name(),
+                                                                                     static_cast<int>(o->function()->functionType()),
+                                                                                     o->result().toDouble(), this));
+        }
+
+        break;
+      }
+      case UtilityTraceResultObjectType::UtilityGeometryTraceResult:
+      {
+        // Trace completed with Geometry Result
+        setIsResetResultsEnabled(true);
+        auto geometryTraceResult = static_cast<UtilityGeometryTraceResult*>(result);
+
+        auto multipoint = geometryTraceResult->multipoint();
+        if (!multipoint.isEmpty())
+        {
+          SimpleMarkerSymbol* resultPointSymbol = new SimpleMarkerSymbol(SimpleMarkerSymbolStyle::Circle,
+                                                                         QColor(0, 0, 255, 126),
+                                                                         20,
+                                                                         this);
+          auto graphic = new Graphic(multipoint,
+                                     resultPointSymbol,
+                                     this);
```

Comment:
these will be a space leak if you ever clear the graphics overlay unless you explicitly delete them all